lib/customer_io.dart

```diff
@@ -8,23 +10,70 @@ import 'messaging_in_app/platform_interface.dart';
 import 'messaging_push/platform_interface.dart';
 
 class CustomerIO {
-  const CustomerIO._();
+  static CustomerIO? _instance;
+
+  final CustomerIOPlatform _platform;
+  final CustomerIOMessagingPushPlatform _pushMessaging;
+  final CustomerIOMessagingInAppPlatform _inAppMessaging;
+
+  /// Private constructor to enforce singleton pattern
+  CustomerIO._({
+    CustomerIOPlatform? platform,
+    CustomerIOMessagingPushPlatform? pushMessaging,
+    CustomerIOMessagingInAppPlatform? inAppMessaging,
+  })  : _platform = platform ?? CustomerIOPlatform.instance,
+        _pushMessaging =
+            pushMessaging ?? CustomerIOMessagingPushPlatform.instance,
+        _inAppMessaging =
+            inAppMessaging ?? CustomerIOMessagingInAppPlatform.instance;
+
+  /// Get the singleton instance of CustomerIO
+  static CustomerIO get instance {
+    if (_instance == null) {
+      throw StateError(
+        'CustomerIO SDK must be initialized before accessing instance.\n'
+        'Call CustomerIO.initialize() first.',
+      );
+    }
+    return _instance!;
+  }
+
+  /// For testing: create a new instance with mock implementations
+  @visibleForTesting
+  static CustomerIO createInstance({
+    CustomerIOPlatform? platform,
+    CustomerIOMessagingPushPlatform? pushMessaging,
+    CustomerIOMessagingInAppPlatform? inAppMessaging,
+  }) {
+    _instance = CustomerIO._(
+      platform: platform,
+      pushMessaging: pushMessaging,
+      inAppMessaging: inAppMessaging,
+    );
+    return _instance!;
+  }
 
-  static CustomerIOPlatform get _customerIO => CustomerIOPlatform.instance;
+  @visibleForTesting
+  static void reset() {
+    _instance = null;
+  }
 
-  static CustomerIOMessagingPushPlatform get _customerIOMessagingPush =>
-      CustomerIOMessagingPushPlatform.instance;
+  /// Access push messaging functionality
+  CustomerIOMessagingPushPlatform get pushMessaging => _pushMessaging;
 
-  static CustomerIOMessagingInAppPlatform get _customerIOMessagingInApp =>
-      CustomerIOMessagingInAppPlatform.instance;
+  /// Access in-app messaging functionality
+  CustomerIOMessagingInAppPlatform get inAppMessaging => _inAppMessaging;
 
   /// To initialize the plugin
   ///
   /// @param config includes required and optional configs etc
-  static Future<void> initialize({
-    required CustomerIOConfig config,
-  }) {
-    return _customerIO.initialize(config: config);
+  static Future<void> initialize({required CustomerIOConfig config}) async {
```

Comment:
this is still static function, because `CustomerIO.instance.initialize` didn't look good and then I checked what others could be doing, and saw Firebase also has initialize method as static while others instance bound.